cli/src/pcluster/validators/networking_validators.py

```diff
@@ -47,6 +47,15 @@ def _validate(self, subnet_ids: List[str]):
                         FailureLevel.ERROR,
                     )
 
+            # warn if subnets are not in the same AZ
+            azs = list(dict.fromkeys([subnet.get("AvailabilityZone") for subnet in subnets]))
+            if len(azs) > 1:
+                self._add_failure(
+                    f"Subnets in multiple Availibility Zones {azs} may result in data transfer charges. "
+                    + "Please review the EC2 docs: https://aws.amazon.com/ec2/pricing/on-demand/#Data_Transfer",
+                    FailureLevel.WARNING,
+                )
```

Comment:
I'm ok with moving this check from the schema level into a suppressable validator however we can't mark this as a warning until the shared storage functionalities are also made compatible with multi az. I suggest to keep this as an error for now and re-word the error message.

Reply:
What's the issue with shared storage?